File: frontend/src/api/ai.ts
import apiClient from "./client";

export interface AIParsedSchedule{
    title: string;
    date: string; // YYYY-MM-DD
    content: string | null;
    start_time: string; // HH:MM
    end_time: string; // HH:MM
}

export interface AIParseResponse{
    is_complete: boolean;
    data: AIParsedSchedule | null; // is_complete가 true일 경우 ScheduleData 객체
    question: string | null; // is_complete가 false일 경우 사용자에게 던질 질문
}

/**
 * AI 파싱 엔드포인트에 텍스트를 전송학 응답을 받습니다.
 * @param text 파싱할 자연어 텍스트
 * @returns AI 파싱 결과
 */
export const parseTextWithAI = async (text: string): Promise<AIParseResponse> => {
    try {
        const response = await apiClient.post<AIParseResponse>('/ai/parse', {text});
        return response.data;
    } catch (error) {
        console.error('AI에게 요청한 문자 파싱 작업 중 에러: ', error);
        throw error;
    }
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field
from datetime import date as DateType, time as TimeType

class AIParsedSchedule(BaseModel):
    """
    AI에 의해 파싱된 일정 정보 응답 스키마
    Modelfile의 시스템 프롬프트에서 정의한 JSON 형식과 일치해야 합니다.
    """
    title: str = Field(..., description="파싱된 일정 제목")
    date: DateType = Field(..., description="파싱된 날짜 (YYYY-MM-DD)")
    content: str | None = Field(None, description="파싱된 내용")
    start_time: TimeType = Field(..., description="파싱된 시작 시간 (HH:MM)")
    end_time: TimeType = Field(..., description="파싱된 종료 시간 (HH:MM)")

class AITextRequest(BaseModel):
    """ AI 텍스트 파싱을 위한 요청 스키마"""
    text: str = Field(..., min_length=1, description="AI가 파싱할 자연어 텍스트")
    
class AIParseResponse(BaseModel):
    """AI 파싱 결과 응답"""
    is_complete: bool = Field(...,description="일정 정보 파싱이 완료되었는지 여부")
    data: AIParsedSchedule | None = Field(None, description="파싱 완료된 일정 데이터 (is_complete가 false이면 null)")
    question: str | None = Field(None, description="정보가 부족하여 AI가 되묻는 질문 (is_complete가 true이면 null)")
